_posts/2020-02-07-metal3-dev-env-install-deep-dive.md

```diff
@@ -0,0 +1,921 @@
+<br>
+<br>
+
+> * **Libvirt.**
+
+>It actually is the role that configures the virtual bare metal servers. They are all identically defined with the same hardware and network configuration. Note that they are not started since they will be booted later by ironic during the provisioning process.  
+
+<br>
+> **Note:**  It is possible to change the number of VMs to provision by replacing the value of `NUMBER_NODES`
+
+<br>
+
+> Finally, once the VMs are defined and we have their MAC address, the ironic inventory file `ironic_nodes_json` is created. The action of creating a node is part of the enrollment process and the first step to prepare a node to reach the "available" status.
+
+```json
+{
+  "nodes": [
+    {
+      "name": "node-0",
+      "driver": "ipmi",
+      "resource_class": "baremetal",
+      "driver_info": {
+        "username": "admin",
+        "password": "password",
+        "port": "6230",
+        "address": "ipmi://192.168.111.1:6230",
+        "deploy_kernel": "http://172.22.0.1/images/ironic-python-agent.kernel",
+        "deploy_ramdisk": "http://172.22.0.1/images/ironic-python-agent.initramfs"
+      },
+      "ports": [
+        {
+          "address": "00:00:e0:4b:24:8b",
+          "pxe_enabled": true
+        }
+      ],
+      "properties": {
+        "local_gb": "50",
+        "cpu_arch": "x86_64"
+      }
+    },
+    {
+      "name": "node-1",
+      "driver": "ipmi",
+      "resource_class": "baremetal",
+      "driver_info": {
+        "username": "admin",
+        "password": "password",
+        "port": "6231",
+        "address": "ipmi://192.168.111.1:6231",
+        "deploy_kernel": "http://172.22.0.1/images/ironic-python-agent.kernel",
+        "deploy_ramdisk": "http://172.22.0.1/images/ironic-python-agent.initramfs"
+      },
+      "ports": [
+        {
+          "address": "00:00:e0:4b:24:8f",
+          "pxe_enabled": true
+        }
+      ],
+      "properties": {
+        "local_gb": "50",
+        "cpu_arch": "x86_64"
+      }
+    },
+...
+...
+...
+```
+
+<br>
+
+> **Information:** This role is also used to tear down the virtual infrastructure depending on the variable [libvirt_action](https://github.com/metal3-io/metal3-dev-env/blob/2b5d8e76f33d143757d1f0b9b1e82dc662245b9c/vm-setup/roles/libvirt/defaults/main.yml#L2) inside the Ansible role: `setup or teardown`.
+
+<br>
+
+> * **Virtbmc**
+
+> This role is only executed if the bare metal virtual machines are created in libvirt, because `vbmc` needs libvirt to emulate a real BMC.
+<br>
+<br>
+> **Information:**  VirtualBMC (`vmbc`) tool simulates a Baseboard Management Controller (BMC) by exposing IPMI responder to the network and talking to libvirt at the host vBMC is running at. Basically, manipulate virtual machines which pretend to be bare metal servers.
```

Comment:
> **Information:**  the VirtualBMC (`vmbc`) tool simulates..

it also needs a blank line after and before the "Information" sentence, otherwise doesn't show as a different content from the normal text.